src/ngamsCore/ngamsLib/ngamsDbCore.py

```diff
@@ -454,7 +454,10 @@ def __init__(self, pool, query, args):
         try:
             self.conn = pool.connection()
             self.cursor = self.conn.cursor()
-            self.cursor.execute(query, args)
+            if args:
```

Comment:
nit-pick: could you copy over the comment that explains the need for this `if` from `transaction.execute`? Even better, if you could put this duplicated logic with a comment in a new non-public free-standing function that'd be great.